### DISCUSSION TITLE
What is the difference between hide and unmount?

### DISCUSSION THREAD

**Discussion by MLFlexer** [MunifTanjim/nui.nvim#316] [Q&A] [answered]
I'm having a hard time understanding the conceptual difference between hide and unmount from the layout docs.

When would one use unmount over hide or unmount over hide?

**Comment by MunifTanjim** [comment, marked as answer]
`:hide` doesn't do any cleanup, `:unomunt` does cleanup.

Say you `:mount` a popup, then modify it's content (e.g. write something on the buffer).

If you do `:hide`, that modified buffer content will be preserved. If you do `:show` afterwards, you'll see the same content.

If you do `:unmount`, the buffer will be destroyed (along with other stuffs, e.g. keymaps). If you do `:mount`/`:show` afterwards, a new buffer will be created.

ref: https://github.com/MunifTanjim/nui.nvim/tree/main/lua/nui/popup#popuphide

**Reply by MLFlexer** [reply]
Awesome, this clears it up, thanks!
